app/pages/homepage/ui.test.js

```diff
@@ -54,47 +63,98 @@ test('should navigate to home page header banner', async (t) => {
 
-  const getLocation = ClientFunction(() => document.location.href);
   const loginComponent = Selector('[data-test-id="login-logout-component"] a');
   await t
     .click(loginComponent)
     .expect(getLocation()).eql('http://identity-server/login');
 });
 
 test('when user is authenticated - should display the logout link', async (t) => {
-  await pageSetup(t, true);
+  await pageSetup({ t, withAuth: true, withLoggedInCookies: true });
 
   const logoutComponent = Selector('[data-test-id="login-logout-component"] a');
   await t
     .expect(await extractInnerText(logoutComponent)).eql('Log out');
 });
 
+test('when user is authenticated - should display log in text once log out link is clicked', async (t) => {
+  await pageSetup({ t, withAuth: true, withLoggedInCookies: true });
+
+  const logoutComponent = Selector('[data-test-id="login-logout-component"] a');
+  await t
+    .expect(logoutComponent.exists).eql(true)
+    .click(logoutComponent)
+    .expect(Selector('[data-test-id="login-logout-component"] a').withText('Log in').exists).eql(true)
+    .expect(Selector('[data-test-id="login-logout-component"] a').withText('Log out').exists).eql(false);
+});
+
+test
+  .requestHooks(logger)('when user is authenticated - should delete cookies once log out link is clicked', async (t) => {
+    await pageSetup({ t, withAuth: true, withLoggedInCookies: true });
+
+    const logoutComponent = Selector('[data-test-id="login-logout-component"] a');
+
+    await t
+      .expect(logoutComponent.exists).eql(true).debug()
+      .expect(await getCookies()).contains('cookie1=cookie1')
+      .click(logoutComponent)
+      .expect(logger.contains(r => r.response.statusCode === 200)).ok()
+      .expect(logger.requests[0].request.headers.cookie).eql(undefined);
+  });
+
+test
+  .requestHooks(logger)('when user is authenticated - should navigate to / once log out link is clicked', async (t) => {
+    await pageSetup({ t, withAuth: true, withLoggedInCookies: true });
+
+    const logoutComponent = Selector('[data-test-id="login-logout-component"] a');
+
+    await t
+      .expect(logoutComponent.exists).eql(true).debug()
+      .expect(await getCookies()).contains('cookie1=cookie1')
+      .click(logoutComponent)
+      .expect(getLocation()).eql('http://localhost:1234/');
+  });
+
+test('when user is authenticated - should navigate to the identity server log in page when clicking the log in link after logging out', async (t) => {
+  await pageSetup({ t, withAuth: true, withLoggedInCookies: true });
+  nock('http://identity-server')
+    .get('/login')
+    .reply(200);
+
+  const logoutComponent = Selector('[data-test-id="login-logout-component"] a');
+  await t
+    .expect(logoutComponent.exists).eql(true)
+    .click(logoutComponent).debug()
+    .expect(Selector('[data-test-id="login-logout-component"] a').withText('Log in').exists).eql(true)
+    .expect(Selector('[data-test-id="login-logout-component"] a').withText('Log out').exists).eql(false)
```

Comment:
can we use `extractInnerText` to check it should say `Log in`

Reply:
this test does that

Reply:
I know just we so we have a single way to check the text of a component. Unless there is a specific reason for doing it like this.